src/main/java/com/google/sps/servlets/FamilyServlet.java

```diff
@@ -34,9 +34,9 @@
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         UserService userService = UserServiceFactory.getUserService();
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        String userID = userService.getCurrentUser().getUserId();
+        String userEmail = userService.getCurrentUser().getEmail();
```

Comment:
Have you verified in documentation that this is a guarantee provided by the UserService, and that the email address is guaranteed to (a) be unique and (b) never change per user?

Checking to make sure that this is a guarantee by the API, and not an assumption based on the way email seems to work. Otherwise, there's a potential for tricky issues down the line.

Even assuming a google account can never change its email and never will be able to, multiple email addresses map to the same email. These are all the same:

- firstname.lastname@gmail.com
- firstnamelastname@gmail.com
- firstnamelastname+anythingaftertheplusisignoredbyemaileservers@gmail.com are all the same email address.

Unless the docs specify otherwise, the email could suddenly start deciding to use any of those, and those would be technically correct emails (the best kind of correct!). But all of them would compare differently in the Datastore.